apps/platform/src/components/dashboard/project/editProjectSheet/index.tsx

```diff
@@ -12,19 +13,162 @@ import { Button } from '@/components/ui/button'
 import { Label } from '@/components/ui/label'
 import { Input } from '@/components/ui/input'
 import { Switch } from '@/components/ui/switch'
-import { editProjectOpenAtom } from '@/store'
+import { editProjectOpenAtom, selectedProjectAtom } from '@/store'
+import { ProjectController } from '@keyshade/api-client'
+
+interface FormData {
+  name: string
+  description: string
+  storePrivateKey: boolean
+}
 
 export default function EditProjectSheet(): JSX.Element {
   const [isEditProjectSheetOpen, setIsEditProjectSheetOpen] =
     useAtom(editProjectOpenAtom)
+  const [selectedProject] = useAtom(selectedProjectAtom)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const [formData, setFormData] = useState<FormData>({
+    name: '',
+    description: '',
+    storePrivateKey: false
+  })
+
+  const [initialData, setInitialData] = useState<FormData>({
```

Comment:
We won't need 2 data objects. We can set the initial data in the form data itself.